mopidy/config/types.py

```diff
@@ -178,30 +219,117 @@ def serialize(self, value, display=False):
             raise ValueError(f"{value!r} is not a boolean")
 
 
+class Pair(ConfigValue):
+    """Pair value
+
+    The value is expected to be a pair of elements, separated by a specified delimiter.
+    Values can optionally not be a pair, in which case the whole input is provided for
+    both sides of the value.
+    """
+
+    def __init__(
+        self, optional=False, optional_pair=False, separator="|", subtypes=None
+    ):
+        self._required = not optional
+        self._optional_pair = optional_pair
+        self._separator = separator
+        if subtypes:
+            self._subtypes = subtypes
+        else:
+            self._subtypes = (String(), String())
+
+    def deserialize(self, value):
+        raw_value = decode(value).strip()
+        validators.validate_required(raw_value, self._required)
+        if not raw_value:
+            return None
+
+        if self._separator in raw_value:
+            values = raw_value.split(self._separator, 1)
+        elif self._optional_pair:
+            values = (raw_value, raw_value)
+        else:
+            raise ValueError("must have separator")
+
+        return (
+            self._subtypes[0].deserialize(encode(values[0])),
+            self._subtypes[1].deserialize(encode(values[1])),
+        )
+
+    def serialize(self, value, display=False):
+        serialized_first_value = self._subtypes[0].serialize(
+            value[0], display=display
+        )
+        serialized_second_value = self._subtypes[1].serialize(
+            value[1], display=display
+        )
+
+        if (
+            not display
+            and self._optional_pair
+            and serialized_first_value == serialized_second_value
+        ):
+            return serialized_first_value
+        else:
+            return "{0}{1}{2}".format(
+                serialized_first_value,
+                self._separator,
+                serialized_second_value,
+            )
+
+
 class List(ConfigValue):
     """List value.
 
-    Supports elements split by commas or newlines. Newlines take presedence and
+    Supports elements split by commas or newlines. Newlines take precedence and
     empty list items will be filtered out.
     """
 
-    def __init__(self, optional=False):
+    def __init__(self, optional=False, unique=False, subtype=None):
         self._required = not optional
+        self._unique = unique
+        if subtype:
+            self._subtype = subtype
+        else:
+            self._subtype = String()
```

Comment:
I think this reads a little bit quicker:

```suggestion
        self._subtype = subtype if subtype else String()
```

Reply:
I was burned so many times by PHP's crappy ternary operator that I still forget Python has non-crap ternary statements. I agree this is better here :+1:

Reply:
I've made this change :+1: